types/diagnostic.d.ts

```diff
@@ -10,5 +11,5 @@ export interface ScriptErrorData {
 
 export interface ImageErrorData {
     source: string;
-    target: number;
+    target: Target;
```

Comment:
>Target [](start = 12, length = 6)

This is mixing the typing for the wire data and the in-memory representation. Ideally you'd separate them into two different types if they don't agree.

For now you could at least make this number | Target and add a comment why the union type is needed.